### ISSUE TITLE
Terminal interaction breaks for @clack/prompts

### ISSUE THREAD

**Issue by Subwaytime** [oven-sh/bun#4835]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.2.0-32-generic x86_64 x86_64

### What steps can reproduce the bug?

1. Install `@clack\prompts
```
bun add @clack\prompts
```
2. Initialize Bun
```
bun init
```
3. Add some code
```ts
import { intro, text, select, outro } from '@clack/prompts';

intro('Bun Test');

const name = await text({
    message: 'Choose a name'
});

const type = await select({
	message: 'Pick a project type.',
	options: [
		{ value: 'ts', label: 'TypeScript' },
		{ value: 'vue', label: 'Vue', hint: '3' },
		{ value: 'laravel', label: 'Laravel' },
	]
});

outro('Bun baked');
```

3. Try selecting any input after a `text prompt` via arrow keys

### What is the expected behavior?

It should return the same results as in node.

![image](https://github.com/oven-sh/bun/assets/7354569/95945c22-d104-4575-80c1-4a22118552ef)

### What do you see instead?

![image](https://github.com/oven-sh/bun/assets/7354569/d98da795-6ff4-418c-b796-2ad77c183876)

### Additional information

Note: Select, Confirm and any other prompt work if its the **ONLY** prompt. If the error appears like in the picture above, its also really hard to kill the Terminal as it seems to lag/hang quite a bit.

**Comment by kkirkfield** [comment]
I get the same issue for Svelte projects.

![image](https://github.com/oven-sh/bun/assets/1903771/34164c51-cc73-45f7-80da-1f86d4fa77f5)

**Comment by Subwaytime** [comment]
Taking a look at  [vitepress](https://github.com/vuejs/vitepress/) - Its a bit odd that these issues occur. 
https://github.com/vuejs/vitepress/blob/main/src/node/init/init.ts 
This code here via `bunx vitepress init` with `@clack\prompts` works perfectly for multiple prompts..

**Comment by Subwaytime** [comment]
I found a workaround for now: **wrapping everything inside a group**.

Based on the code above, it looks something like this:

```ts
import { intro, text, select, outro, group } from '@clack/prompts';

intro('Bun Test');

const test = await group({
    name: () => text({
        message: 'Choose a name'
    }),
    type: () => select({
        message: 'Pick a project type.',
        options: [
            { value: 'ts', label: 'TypeScript' },
            { value: 'vue', label: 'Vue', hint: '3' },
            { value: 'laravel', label: 'Laravel' },
	]
    })
}, {
    onCancel: () => {
        cancel('Operation cancelled.');
        return process.exit(0);
    }
});

outro('Bun baked');
```

**Comment by IPv4Addr** [comment]
> I get the same issue for Svelte projects.
> 
> ![image](https://user-images.githubusercontent.com/1903771/266876840-34164c51-cc73-45f7-80da-1f86d4fa77f5.png)

Somehow for me using:
```
bunx create-svelte {dir) 
```
The keyup / down spacebar navigation started working. 
But not providing any dir as an argument does not work?
Hope this helps!

**Comment by manzt** [comment]
Thanks for the group tip. I made this proxy helper for the group hack:

```ts
import * as prompts from "@clack/prompts";

let p = new Proxy(prompts, {
  get(target, prop) {
    if (prop == "select" || prop == "text") {
      let fn = target[prop];
      return async opts => {
        let { value } = await p.group({ value: () => fn(opts) });
        return value;
      };
    }
    return Reflect.get(target, prop);
  }
});
```

```ts
p.intro("Bun Test");

const name = await p.text({
  message: "Choose a name"
});

const type = await p.select({
  message: "Pick a project type.",
  options: [
    { value: "ts", label: "TypeScript" },
    { value: "vue", label: "Vue", hint: "3" },
    { value: "laravel", label: "Laravel" }
  ]
});

p.outro("Bun baked");
```

**Comment by nektro** [comment]
does this still reproduce for you? it appears to be working on my end in Bun 1.1.10 (you may need to run `bun upgrade`)

**Comment by Subwaytime** [comment]
Will do a check and report back!

**Comment by Subwaytime** [comment]
Seems to be resolved for Bun `1.1.10`!
